### ISSUE TITLE
Thermostat state not detected

### ISSUE THREAD

**Issue by kalhimeo** [nervetattoo/simple-thermostat#169]
Dear developer,

Thank you very much for this nice lovelace card. I think that I am facing a bug which makes your card not to detect the actual state of the thermostat entity.
I have some KNX thermostats for which I can see the state in the HA developer tool (those are either 'off' or 'heat') and I wanted to have a different icon for both cases so I added this to the card :

```
icon:
  heat: 'mdi:radiator'
  'off': 'mdi:radiator-off'
```

But unfortunately that causes the card not to show any icon. By troubleshooting the problem, I realized that the card is also supposed to show the 'state' as text by default ('hide state' is false by default) but in my case the state is not shown in the card even without configuring to hide it.

So my conclusion is that somehow the card is not able to read the state of my climate entities which are created from the KNX integration. I am using HA 117.0. For reference, here is the list of attributes of one of such thermostat :

```
hvac_modes:
  - heat
  - 'off'
min_temp: 18.5
max_temp: 24.5
target_temp_step: 0.5
preset_modes:
  - away
  - sleep
  - comfort
  - eco
current_temperature: 20.8
temperature: 21.5
preset_mode: comfort
friendly_name: Living
supported_features: 17
```

I hope that you can find the cause as your card is really great :-)

Thanks in advance.

**Comment by KayGt3** [comment]
Hello @kalhimeo ,

here below you can find my card configuration, i hope it can be helpful.

```
type: 'custom:simple-thermostat'
entity: climate.netatmo_corridoio
step_layout: row
name: false
icon: true
control:
  hvac:
    'off': 
       name: Off
       icon: mdi:power
```

**Comment by kalhimeo** [comment]
Hi KayGt3,

Thanks for the config, but actually I have the card working, it's just the state of the thermostat which is not showing. Here is an example of my card, but I tried with only basic config as well, in both case I cannot show the state and the main icon related to it (top left) does not show at all :

```
type: 'custom:simple-thermostat'
entity: climate.thermostat_salon
name: Bureau
step_layout: row
toggle_entity: switch.chauffage_inoccupation_commun
hide:
  temperature: true
icon:
  heat: 'mdi:radiator'
  'off': 'mdi:radiator-off'
label:
  temperature: Pièce
control:
  _headings: false
  _names: false
  preset:
    _hide_when_off: false
    away:
      icon: 'mdi:account-cancel-outline'
      name: Absent
    sleep:
      icon: 'mdi:leaf'
      name: Éco
    comfort:
      icon: 'mdi:radiator'
      name: Normal
    eco:
      icon: 'mdi:snowflake-off'
      name: Hors Gel
sensors:
  - entity: sensor.temperature_bureau_moyenne
    name: T° Pièce
    unit: ' '
  - entity: sensor.valve_chauffage_bureau
    name: Vanne
```

**Comment by KayGt3** [comment]
I think you need to play around "control:" section on the card.

**Comment by kalhimeo** [comment]
Hi KayGt3,

Thanks for trying to help, but I don't think it's related, even something as simple as this does not show the state :
```
type: 'custom:simple-thermostat'
entity: climate.thermostat_salon
hide:
  state: false
```

**Compare my result and the one from the documentation, the state is missing (disregard the different theme) :**

![Capture d’écran 2020-11-29 142602](https://user-images.githubusercontent.com/6355373/100543184-f4c01900-324e-11eb-8344-99dfd94c9886.png)

![Capture d’écran 2020-11-29 142513](https://user-images.githubusercontent.com/6355373/100543185-f558af80-324e-11eb-8525-073eb176415e.png)

**However my entity has the state "heat" :**

![Capture d’écran 2020-11-29 142904](https://user-images.githubusercontent.com/6355373/100543314-ab23fe00-324f-11eb-891c-c73b12d1f352.png)

**Comment by KayGt3** [comment]
Try to add "hvac_modes:" under "control:" like below

```
control:
  hvac_modes:
    'off':
    heat:
```
Also have a look here (maybe it helps): https://github.com/nervetattoo/simple-thermostat#usage-of-the-control-config

**Comment by kalhimeo** [comment]
Problem fixed with the new 0.40 release, thanks @tomvanswam for implementing #180 !
